source/_ext/build_engine_schema_docs.py

```diff
@@ -0,0 +1,193 @@
+import os
+from typing import Any, Dict
+
+from howso.client.api import get_api
+from jinja2 import BaseLoader, Environment
+from jinja2.exceptions import UndefinedError
+from sphinx.util import logging
+
+logger = logging.getLogger(__name__)
+
+
+def setup(app):
+    """
+    Setup function for the Sphinx extension.
+    This builds RST pages for the needed Engine type schemas.
+    """
+    def generate_rst_from_json(app):
+        """
+        Function that runs before Sphinx starts reading source files.
+        Place your JSON fetching and RST generation code here.
+        """
+        try:
+            needed_schemas = [
+                "FeatureAttributes",
+                "FeatureBounds",
+                "FeatureDataType",
+                "FeatureOriginalType",
+                "FeatureOriginalTypeBoolean",
+                "FeatureOriginalTypeDate",
+                "FeatureOriginalTypeDatetime",
+                "FeatureOriginalTypeInteger",
+                "FeatureOriginalTypeNumeric",
+                "FeatureOriginalTypeObject",
+                "FeatureOriginalTypeString",
+                "FeatureOriginalTypeTime",
+                "FeatureOriginalTypeTimedelta",
+                "FeatureType",
+            ]
+
+            # THIS WILL NOT WORK IF HOWSO-ENGINE-PY IS INSTALLED EDITABLY
+            # (unless you have the .caml in your local repo too)
+            schema_map = get_api()['schemas']
+            os.makedirs("howso/types", exist_ok=True)
```

Comment:
This needs to generate relative to the source directory of the app. Running `make html` locally generates this in the wrong location

Reply:
Looks like you can use `app.srcdir`